cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/condition/PipelineCondition.java

```diff
@@ -16,17 +16,93 @@
 
 package co.cask.cdap.etl.batch.condition;
 
-import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.macro.MacroEvaluator;
+import co.cask.cdap.api.metrics.Metrics;
+import co.cask.cdap.api.plugin.PluginContext;
+import co.cask.cdap.api.workflow.AbstractCondition;
 import co.cask.cdap.api.workflow.WorkflowContext;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import co.cask.cdap.etl.api.condition.Condition;
+import co.cask.cdap.etl.api.condition.ConditionContext;
+import co.cask.cdap.etl.batch.BatchPhaseSpec;
+import co.cask.cdap.etl.common.BasicArguments;
+import co.cask.cdap.etl.common.Constants;
+import co.cask.cdap.etl.common.DefaultMacroEvaluator;
+import co.cask.cdap.etl.common.PipelinePhase;
+import co.cask.cdap.etl.common.SetMultimapCodec;
+import co.cask.cdap.etl.common.plugin.PipelinePluginContext;
+import co.cask.cdap.etl.spec.StageSpec;
+import co.cask.cdap.internal.io.SchemaTypeAdapter;
+import com.google.common.collect.SetMultimap;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 
+import java.util.HashMap;
+import java.util.Map;
 import javax.annotation.Nullable;
 
 /**
  * Represents conditions in the data pipelines.
  */
-public class PipelineCondition implements Predicate<WorkflowContext> {
+public class PipelineCondition extends AbstractCondition {
+
+  // This is only visible during the configure time, not at runtime.
+  private final BatchPhaseSpec phaseSpec;
+  private Metrics metrics;
+
+  private static final Gson GSON = new GsonBuilder()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .registerTypeAdapter(SetMultimap.class, new SetMultimapCodec<>())
+    .create();
+
+  public PipelineCondition(BatchPhaseSpec spec) {
+    this.phaseSpec = spec;
+  }
+
+  @Override
+  protected void configure() {
+    setName(phaseSpec.getPhaseName());
+    setDescription("Condition phase executor. " + phaseSpec.getPhaseName());
+
+    // add source, sink, transform ids to the properties. These are needed at runtime to instantiate the plugins
+    Map<String, String> properties = new HashMap<>();
+    properties.put(Constants.PIPELINEID, GSON.toJson(phaseSpec));
+    setProperties(properties);
+  }
+
   @Override
   public boolean apply(@Nullable WorkflowContext input) {
-    return false;
+    if (input == null) {
+      // should not happen
+      throw new IllegalStateException("WorkflowContext for the Condition cannot be null.");
+    }
+    Map<String, String> properties = input.getConditionSpecification().getProperties();
```

Comment:
I realized that we need an api change for the WorkflowContext to get the ConditionSpecifications. Have added it in this PR.